arangod/Aql/Query.cpp

```diff
@@ -497,6 +497,10 @@ ExecutionPlan* Query::preparePlan() {
 
     _ast->validateAndOptimize();
 
+    if(!trx->transactionContextPtr()->getParentTransaction()){
```

Comment:
We should rename this hint.... 
`FROM_AQL` is only part of the truth here, as the transaction is start from AQL *and is also a top-level transaction*.
This should be reflected in the hint's name.
Additionally, the hint should be set before `transaction::Methods::begin()` is called on the transaction, because that's how it is done with all other hints.

Reply:
Ok, it is.